File: corenlp.java
import okhttp3.*;

public class CoreNLPExample {
   public static void main(String[] args) throws java.io.IOException {
   // The text to annotate
   String text = "The fox jumped over the lazy dog";

   // Create an OkHttp Client with authentication
   OkHttpClient client = new OkHttpClient.Builder()
      .authenticator((route, response) -> {
         String credential = Credentials.basic("api_key", "api_secret");
         return response.request().newBuilder().header("Authorization", credential).build();
         })
      .build();

   // Issue a request
   Request request = new Request.Builder()
      .url("https://corenlp.ai")
      .post(RequestBody.create(MediaType.parse("text/plain; charset=utf-8"), text))
      .build();

      // Read the JSON response
      Response response = client.newCall(request).execute();
      String json = response.body().string();

      System.out.println(json);
   }
}

File: corenlp.rb
require 'net/http'
require 'json'

uri = URI('https://corenlp.ai/')

Net::HTTP.start(uri.host, uri.port,
   :use_ssl => uri.scheme == 'https') do |http|

   # Create the request
   request = Net::HTTP::Post.new(uri.request_uri)
   request.body = 'the fox jumped over the lazy dog'
   request.basic_auth 'api_key', 'api_secret'

   # Issue the request
   response = http.request(request)

   # Parse the annotated text
   annotated_text = JSON.parse(response.body)
   puts annotated_text
end
